dispatches/models/fossil_case/ultra_supercritical_plant/storage/charge_design_ultra_supercritical_power_plant.py

```diff
@@ -0,0 +1,2672 @@
+from idaes.models.unit_models import (HeatExchanger,
+                                      MomentumMixingType,
+                                      Heater,
+                                      Mixer,
+                                      PressureChanger)
+from idaes.models.unit_models.separator import (Separator,
+                                                SplittingType)
+from idaes.models.unit_models.heat_exchanger import (delta_temperature_underwood_callback,
+                                                     HeatExchangerFlowPattern)
+from idaes.models.unit_models.pressure_changer import ThermodynamicAssumption
+from idaes.models_extra.power_generation.unit_models.helm import (HelmMixer,
+                                                                  HelmIsentropicCompressor,
+                                                                  HelmTurbineStage,
+                                                                  HelmSplitter)
+
+# Import ultra-supercritical power plant model
+from dispatches.models.fossil_case.ultra_supercritical_plant import (
+    ultra_supercritical_powerplant as usc)
+
+# Import properties package for storage materials
+from dispatches.models.fossil_case.properties import (solarsalt_properties,
+                                                      hitecsalt_properties,
+                                                      thermaloil_properties)
+
+scaling_obj = 1e-7
+
+def create_charge_model(m, add_efficiency=None, power_max=None):
+    """Create flowsheet and add unit models
+
+    """
+
+    # Create a block to add charge storage model
+    m.fs.charge = Block()
+
+    # Add model data
+    _add_data(m)
+
+    # Add molten salt and thermal oil properties
+    m.fs.solar_salt_properties = solarsalt_properties.SolarsaltParameterBlock()
+    m.fs.hitec_salt_properties = hitecsalt_properties.HitecsaltParameterBlock()
+    m.fs.therminol66_properties = thermaloil_properties.ThermalOilParameterBlock()
+
+    ###########################################################################
+    #  Add unit models
+    ###########################################################################
+
+    # Add connector, defined as a dummy heat exchanger with Q=0 and a
+    # deltaP=0
+    m.fs.charge.connector = Heater(
+        default={
+            "dynamic": False,
+            "property_package": m.fs.prop_water,
+            "has_pressure_change": False
+        }
+    )
+
+    # Add cooler to ensure the outlet of charge heat exchanger is a
+    # subcooled liquid before mixing it with the plant
+    m.fs.charge.cooler = Heater(
+        default={
+            "dynamic": False,
+            "property_package": m.fs.prop_water,
+            "has_pressure_change": True
+        }
+    )
+
+    # Add pump to increase the pressure of the water to allow mixing
+    # it at a desired location within the plant
+    m.fs.charge.hx_pump = PressureChanger(
```

Comment:
Is there a reason for using generic pump instead of `HelmIsentropicCompressor`?

Reply:
generic pressure changer is used to access the costing block